src/Entity/CaseEntity.php

```diff
@@ -686,4 +670,6 @@ public function setRemovalReason(?string $removalReason): self
 
         return $this;
     }
+
+    abstract public function getNonRelevantComplainantPropertiesWithRespectToValidation(): array;
```

Comment:
Add documentation on what this function is used for and what it must return.

I think it will make more sense to list properties that require re-validation of an identification rather than listing properties that *does not* influence the identification. It may also be useful (when implementing `EventSubscriberInterface`) to tell which identification is affected, i.e. using something like

```php
function getIdentificationDependencies(): array {
  return [
    'complainantIdentification' => [
      // Depends on values of
      'name', 
      'address',
    ],
    'anotherIdentification' => [
      // Depends on values of
      'accuserName', 
      'accuserAddress',
    ],
  ];
}
```

This way we know that if `name` is changed the `complainantIdentification` must be invalidated.

Reply:
Agreed. Listing the properties that should result in re-validation instead. This also significantly makes the code more readable.